utils/metrics.go

```diff
@@ -21,11 +21,21 @@ import (
 	"sync"
 )
 
+// ReporterType is the type of reporter
+type ReporterType int
+
+// List of reporter types
+const (
+	ReporterTypeDataNode ReporterType = iota
```

Comment:
shall we add subscriber and controller too?

Reply:
subscriber and controller don't use this di yet.
we can easily add new types and sentinels when we migrate them to this